functions/index.js

```diff
@@ -1,8 +1,40 @@
 const functions = require('firebase-functions');
+const admin = require('firebase-admin');
 
-// // Create and Deploy Your First Cloud Functions
-// // https://firebase.google.com/docs/functions/write-firebase-functions
-//
-exports.helloWorld = functions.https.onRequest((request, response) => {
-  response.send('Hello from Firebase!');
-});
+admin.initializeApp();
+const db = admin.firestore();
+const fieldValue = admin.firestore.FieldValue;
+
+exports.onFollow = functions.firestore
+  .document('/users/{userId}/followers/{followerId}')
+  .onCreate((snapshot, context) => {
+    const { userId } = context.params;
+    const currentUser = snapshot.data().ref;
+    return db.collection('users').doc(currentUser.id).collection('followed')
+      .add({
+        ref: db.collection('users').doc(userId),
```

Comment:
```suggestion
        ref: db.collection('users').doc(snapshot.parent.parent.ref),
```
Pour passer la ref et pas juste l'id

Reply:
Dear Hugo,
i don't get your point, bc it's already inserting the ref of the user, get the doc by his ref or his id is sending the ref anyway no ? 
Regards, 
Your favorite teammate

Reply:
```suggestion
        ref: snapshot.parent.parent.ref,
```
C'est ça que je voulais dire mais c'est pas bien grave laisse comme ça

Reply:
Oh, you're fully right then